### ISSUE TITLE
Upgrade to Angular 5

### ISSUE THREAD

**Issue by DavideViolante** [DavideViolante/Angular-Full-Stack#124]
I'm upgrading the project to Angular 5 but I encountered some unexpected problems.

Branch: https://github.com/DavideViolante/Angular-Full-Stack/compare/upgradev5?expand=1

Error on building:
```[1] ERROR in client/app/services/auth.service.ts(26,53): error TS2339: Property 'map' does not exist on type 'Observable<any>'.```

Any help?

**Comment by mushinz** [comment]
I have updated to angular 5, and i didnt get that error

**Comment by Thavarajan** [comment]
import the map operator

`import 'rxjs/add/operator/map';`

**Comment by Paul75** [comment]
It works if I change into 5 on master ? not have errors.

It seems problems to migrate to Angular 5 ?

**Comment by pedroraft** [comment]
#130 Will fix this; solution is as mentioned by @Thavarajan

**Comment by DavideViolante** [comment]
Seems like I cannot login anymore after the fix. You too?

**Comment by pedroraft** [comment]
True, server is throwing a 403. Very weird, i will look in to it.

**Comment by pedroraft** [comment]
@DavideViolante the request body wasn't hitting the server, the cause is the JSON.stringify(credentials) , simply removing solves the problem. it's unnecessary with httpclient. [This guy](http://brianflove.com/2017/07/21/migrating-to-http-client/) describes very well how to migrate to httpclient.

```typescript
  login(credentials): Observable<any> {
    return this.http.post('/api/login', credentials);
  }
```

Want me to make a pr fixing this in the entire project?

**Comment by DavideViolante** [comment]
You can include the fix in the #130, thanks :+1:

**Comment by pedroraft** [comment]
#130 just turned into a biggy, had to create models in angular, i think is for the best anyway, it was a missing thing; i recommend testing everything, all components went through changes.
